packages/core/lib/utils.js

```diff
@@ -75,7 +75,7 @@ exports.placeholdify = (config) => {
 exports.environmentalize = (_config) => {
   const _env = {};
   const env = global._env || process.env;
-  const regExp = /\[(\w+?)\]/g;
+  const regExp = /\[([a-zA-Z_][a-zA-Z0-9_]*)(?:=(.+))?\]/g;
```

Comment:
This should be `=(.*?)` otherwise `[TEST=]` would not match the regex, which is wrong in my opinion.
And we should go for the `lazy` mode (instead of `greedy`) otherwise `[FOO=foo][BAR=bar]` would also not work.

See https://regex101.com/r/BuXyeU/1 vs https://regex101.com/r/WZhMgw/1

Reply:
Good catch! We definitely have to add unit tests here before the next major release. 😬